### PR TITLE
Add filter for if a user is an attendee for an event

### PR DESCRIPTION
**Pull request by oleast**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [ ] I have added tests for the code I added
- [x] I have provided documentation for the code I added
- [x] The code is ready to be merged

## Description:

- Add a filter which lets a logged in user filter events they are attendees on.
If the user is not logged in, all events are filtered away.

### CODE REVIEW

File: apps/events/filters.py [hunk 1]
```diff
@@ -17,8 +17,25 @@ class EventDateFilter(django_filters.FilterSet):
...
+        """ User cannot attend events if they are not logged in """
+        if not self.request.user.is_authenticated:
+            return queryset.none()
+
+        user_attendees = Attendee.objects.filter(user=self.request.user)
+        attending_events = [attendee.event.pk for attendee in user_attendees]
+        if value:
+            return queryset.filter(pk__in=attending_events)
+        else:
+            return queryset.exclude(pk__in=attending_events)
```
Comment (torjusti) [inline apps/events/filters.py]: I think we should not differ between not providing this and providing false. That is, to make this return all rows in the case where value is not provided.

File: apps/events/filters.py [hunk 2]
```diff
@@ -17,8 +17,25 @@ class EventDateFilter(django_filters.FilterSet):
     event_end__gte = django_filters.DateTimeFilter(field_name='event_end', lookup_expr='gte')
     event_end__lte = django_filters.DateTimeFilter(field_name='event_end', lookup_expr='lte')
     attendance_event__isnull = django_filters.BooleanFilter(field_name='attendance_event', lookup_expr='isnull')
+    is_attendee = django_filters.BooleanFilter(field_name='attendance_event', method='filter_is_attendee')
     event_type = ListFilter()
 
+    def filter_is_attendee(self, queryset, name, value):
+        """
+        Filter events on if a user can attend them or not.
```
Comment (torjusti) [inline apps/events/filters.py]: Should not this say *Filter events based on if a user is attending them or not.*?

Comment (oleast) [reply apps/events/filters.py]: Yes, that is correct

File: apps/events/filters.py (hunk 1 above)
Comment (oleast) [reply apps/events/filters.py]: I agree

**Comment by oleast** [comment]
@torjusti How about now?

**Pull request merged**